src/parsers/manifestjson.js

```diff
@@ -1,103 +1,43 @@
-import esprima from 'esprima';
 import RJSON from 'relaxed-json';
 import validate from 'schema/validator';
 
 import { getConfig } from 'cli';
 import { MANIFEST_JSON, PACKAGE_EXTENSION } from 'const';
 import log from 'logger';
 import * as messages from 'messages';
+import JSONParser from 'parsers/json';
 import { isToolkitVersionString } from 'schema/formats';
 import { singleLineString } from 'utils';
 
-export default class ManifestJSONParser {
+
+export default class ManifestJSONParser extends JSONParser {
 
   constructor(jsonString, collector, {
-    rJSON=RJSON, selfHosted=getConfig().argv.selfHosted,
+    filename=MANIFEST_JSON, rJSON=RJSON, selfHosted=getConfig().argv.selfHosted,
```

Comment:
I wonder if this should be `relaxedJSON` instead of `rJSON` to make it clearer what it represents.